caom2pipe/caom2pipe/execute_composable.py

```diff
@@ -0,0 +1,1163 @@
+
+    def __init__(self, obs_id, collection, collection_pattern,
+                 fname_on_disk=None):
+        self.obs_id = obs_id
+        self.collection = collection
+        self.collection_pattern = collection_pattern
+        self.fname_on_disk = fname_on_disk
+
+    def get_file_uri(self):
+        return 'ad:{}/{}.gz'.format(self.collection, self.get_file_name())
+
+    def get_file_name(self):
+        return '{}.fits'.format(self.obs_id)
+
+    def get_compressed_file_name(self):
+        return '{}.fits.gz'.format(self.obs_id)
+
+    def get_model_file_name(self):
+        return '{}.fits.xml'.format(self.obs_id)
+
+    def get_prev(self):
+        return '{}_prev.jpg'.format(self.obs_id)
+
+    def get_thumb(self):
+        return '{}_prev_256.jpg'.format(self.obs_id)
+
+    def get_prev_uri(self):
+        return self._get_uri(self.get_prev())
+
+    def get_thumb_uri(self):
+        return self._get_uri(self.get_thumb())
+
+    def get_obs_id(self):
+        return self.obs_id
+
+    def get_log_file(self):
+        return '{}.log'.format(self.obs_id)
+
+    def get_product_id(self):
+        return self.obs_id
+
+    def get_fname_on_disk(self):
+        return self.fname_on_disk
+
+    def is_valid(self):
+        pattern = re.compile(self.collection_pattern)
+        return pattern.match(self.obs_id)
+
+    def _get_uri(self, fname):
+        return 'ad:{}/{}'.format(self.collection, fname)
+
+    @staticmethod
+    def remove_extensions(name):
+        return name.replace('.fits', '').replace('.gz', '').replace('.header',
+                                                                    '')
+
+
+class CaomName(object):
+    """The naming rules for making and decomposing CAOM URIs, all isolated in
+    one class. There are probably OMM assumptions built in, but those will
+    slowly go away :). """
+
+    def __init__(self, uri):
+        self.uri = uri
+
+    def get_file_id(self):
+        return self.uri.split('/')[1].split('.')[0]
+
+    def get_file_name(self):
+        return self.uri.split('/')[1]
+
+    def get_uncomp_file_name(self):
+        return self.get_file_name().replace('.gz', '')
+
+
+class CaomExecute(object):
+    """Abstract class that defines the operations common to all Execute
+    classes."""
+
+    def __init__(self, config, task_type, storage_name, command_name,
+                 cred_param,
+                 cadc_data_client=None, caom_repo_client=None,
+                 meta_visitors=None):
+        self.logger = logging.getLogger()
+        self.logger.setLevel(config.logging_level)
+        formatter = logging.Formatter(
+            '%(asctime)s:%(levelname)s:%(name)-12s:%(lineno)d:%(message)s')
+        for handler in self.logger.handlers:
+            handler.setLevel(config.logging_level)
+            handler.setFormatter(formatter)
+        self.logging_level_param = self._set_logging_level_param(
+            config.logging_level)
+        self.obs_id = storage_name.get_obs_id()
```

Comment:
Because that  creates a wide-spread dependency on the StorageName class, instead of one that's limited to the constructor for CaomExecutor. I think it's the Interface-Segregration Principle, but I can't really remember the name of the thing I'm trying to avoid. Mostly it's "don't call the methods of a member class in another class - that makes for high coupling in code."